perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_groupNames.java

```diff
@@ -37,25 +40,32 @@
 	private static final String FRIENDLY_NAME = "groupNames";
 	private static final String A_U_V_GROUP_NAMES = AttributesManager.NS_USER_ATTR_VIRT + ":" + FRIENDLY_NAME;
 
-
 	@Override
 	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
 		Attribute attribute = new Attribute(attributeDefinition);
 		List<String> groupNames = new ArrayList<>();
+		List<Map<String, Object>> names = sess.getPerunBl().getDatabaseManagerBl().getJdbcPerunTemplate().queryForList("SELECT" +
+			" DISTINCT vos.short_name AS vo_short_name, groups.name AS group_name" +
+			" FROM" +
+			" members" +
+			" JOIN vos ON vos.id = members.vo_id" +
+			" JOIN groups_members ON groups_members.member_id = members.id" +
+			" JOIN groups ON groups_members.group_id = groups.id" +
+			" WHERE" +
+			" groups_members.source_group_status = ?" +
+			" AND members.user_id = ?", MemberGroupStatus.VALID.getCode(), user.getId());
```

Comment:
Please add sorting `` order by vo_short_name, group_name;``. It is not required by any logic, but sometimes the value is displayed to users and alphabetical sort is better. Otherwise same item of the list can be found anywhere, when we ask for value twice (while the list in fact hasn't changed).